In [1]:
import scipy.stats as stats
import numpy as np

In [2]:
# Data Provided in the Contingency Table
observed_data = np.array([[50, 70],
                          [80, 100],
                          [60, 90],
                          [30, 50],
                          [20, 50]])

# 1**. State the Hypotheses:**

In [3]:
# Step 1: State the Hypotheses
# H0: There is no association between the type of device (Smart Thermostat vs. Smart Light) and customer satisfaction level.
# H1: There is an association between the type of device (Smart Thermostat vs. Smart Light) and customer satisfaction level.

# **2. Compute the Chi-Square Statistic:**

In [4]:
# Step 2: Compute the Chi-Square Statistic
chi2_statistic, p_value, dof, expected_data = stats.chi2_contingency(observed_data)

# 3**. Determine the Critical Value:**

Using the significance level (alpha) of 0.05 and the degrees of freedom (which is the number of categories minus 1)


In [5]:
# Step 3: Determine the Critical Value
# Degrees of freedom (dof) = (number of rows - 1) * (number of columns - 1)
# Significance level (alpha) = 0.05
alpha = 0.05
critical_value = stats.chi2.ppf(1 - alpha, dof)
critical_value

9.487729036781154

# **4. Make a Decision:**

In [6]:
# Step 4: Make a Decision
decision = "Reject the null hypothesis" if chi2_statistic > critical_value else "Fail to reject the null hypothesis"

# **Submission Guidelines:**

•	Provide a detailed report of your analysis, including each step outlined in the assignment tasks in a python file.


In [7]:
# Report
print(f"Chi-Square Statistic: {chi2_statistic}")
print(f"Degrees of Freedom: {dof}")
print(f"Critical Value: {critical_value}")
print(f"P-Value: {p_value}")
print(f"Decision: {decision}")

Chi-Square Statistic: 5.638227513227513
Degrees of Freedom: 4
Critical Value: 9.487729036781154
P-Value: 0.22784371130697179
Decision: Fail to reject the null hypothesis


In [8]:
# Interpretation
if chi2_statistic > critical_value:
    print("There is a significant association between the type of device purchased and the customer's satisfaction level.")
else:
    print("There is no significant association between the type of device purchased and the customer's satisfaction level.")

There is no significant association between the type of device purchased and the customer's satisfaction level.


#**Steps**
•	Include all calculations, the Chi-Square statistic, the critical value, and your conclusion.

**Hypotheses:**

H0 (Null Hypothesis): No association between device type and customer satisfaction.

H1 (Alternative Hypothesis): Association exists between device type and customer satisfaction.
Compute Chi-Square Statistic:

Use scipy.stats.chi2_contingency() to calculate the Chi-Square statistic, p-value, degrees of freedom, and expected frequencies.

**Determine Critical Value:**

Calculate using stats.chi2.ppf(1 - alpha, dof) where alpha = 0.05 and dof = degrees of freedom.

**Make a Decision:**

Compare the Chi-Square statistic with the critical value to decide whether to reject the null hypothesis.

**Chi-Square Statistic:** 5.64


Critical Value (at
𝛼
=
0.05
α=0.05):- 9.49


**Decision:** Do not reject the null hypothesis.



# **Conclusion:**

Since the Chi-Square statistic (5.64) is less than the critical value (9.49), we do not reject the null hypothesis. This means there is not enough evidence to suggest a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and customer satisfaction levels